sympy/sets/fancysets.py

```diff
@@ -543,21 +545,31 @@ def __new__(cls, *args):
                 raise ValueError(filldedent('''
     Either the start or end value of the Range must be finite.'''))
 
-        if start.is_infinite:
-            if step*(stop - start) < 0:
-                start = stop = S.One
-            else:
-                end = stop
-        if not start.is_infinite:
-            ref = start if start.is_finite else stop
-            n = ceiling((stop - ref)/step)
-            if n <= 0:
-                # null Range
-                start = end = S.Zero
-                step = S.One
-            else:
-                end = ref + n*step
-        return Basic.__new__(cls, start, end, step)
+        if not step.is_Symbol:
+            if start.is_infinite:
+                if step*(stop - start) < 0:
+                    start = stop = S.One
+                else:
+                    end = stop
+            if not start.is_infinite:
+                ref = start if start.is_finite else stop
+                n = ceiling((stop - ref)/step)
+                if (n <= 0) == True:
+                    # null Range
+                    start = end = S.Zero
+                    step = S.One
+                else:
+                    end = ref + n*step
+        else:
+            end = stop
+        # behave like Python 3 range if PY3 else behave like Python 2 xrange
+        if (end > stop) == True and (stop > start) == True and PY3:
```

Comment:
Maybe we don't need this. Python 3 behavior sounds better, consider that next year we're going to drop Python 2 support.

Reply:
Sure I will remove, `PY3` from the above condition as soon as the PR for removing python2 support is merged into master. In fact a few more changes will be needed to remove `xrange`(used for python2 by `Range`) from other places.

Reply:
Just remove this. SymPy's Range should do the same thing on all Python versions regardless of what xrange does.

Reply:
As from the reviews it is apparent that `Range` should behave like `range`, so I will make the change. However, any deprecation policy we need to follow in this case, because currently `Range` adapts itself to behave like `range` and `xrange`. @smichr Due you want to share any reviews on this topic?

Reply:
@smichr Please update this thread with your views on the above discussion. 
I also want to know, that exactly when we are dropping Python 2 support? I am asking this because, if Python 2 support will go then xrange behaviour will also not work, so may be giving a message to the user is not required while using Range because it is understood that xrange is a part of Python 2 and the user will understand that their code is breaking of the the language drop and not due to a change in Range? Any comments on my thought?

Reply:
We are already implementing py3-round or number parsing (I can't remember which, right now) and not worrying about making the behavior different than py2. @oscarbenjamin comment about "SymPy's Range should do the same thing on all Python versions" is, I believe, the correct sentiment.

If I read this correctly, this has to do with what to display as the ending value? If so, we want to be canonical regardless of what Python does with range.

> can directly remove xrange from Range without any complications

yes. The internals implementation should not matter as long as the same result is obtained.